core/runtime/jsruntime.rs

```diff
@@ -352,6 +353,7 @@ pub struct JsRuntimeState {
   /// It will be retrieved by `exception_to_err_result` and used as an error
   /// instead of any other exceptions.
   pub(crate) validate_import_attributes_cb: ValidateImportAttributesCb,
+  pub(crate) custom_module_evaluation_cb: Option<CustomModuleEvaluationCb>,
```

Comment:
After consideration I think this should be a method on the `ModuleLoader` trait.

Reply:
I think ModuleLoader is doing a little bit too much right now TBH -- it does fetching, transpiling and now we'd be adding evaluation.

Module loading is something like a pipeline and we might want to consider how we can structure the callbacks to be more composable. For example, I might want to stack a URL+file fetching component with a TypeScript compilation transpiler, along with wasm/bytes/url module conversion types.

Right now everything is pretty much jammed into module loaders that aren't really re-usable, but we could be building a library of components instead.

Reply:
I'm not sure how feasible it is - the crux is that we want loading to be as efficient as possible and it's happening in an async context. Which somewhat forces us to do all of that stuff in a single function. I'll be happy to revisit it, but for now I'd like to get this stuff landed. I can leave this evaluation callback as is and not move it to `ModuleLoader`.